aggregator/src/bin/aggregator.rs

```diff
@@ -209,36 +204,45 @@ pub struct HeaderEntry {
 
 /// Options for serving the aggregator API.
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(untagged)]
-pub enum AggregatorApi {
-    ListenAddress {
-        /// Address on which this server should listen for connections to the Janus aggregator API
-        /// and serve its API endpoints, independently from the address on which the DAP API is
-        /// served.
-        listen_address: SocketAddr,
-        /// Resource location at which the DAP service managed by this aggregator api can be found
-        /// on the public internet. Required.
-        public_dap_url: Url,
-    },
-    PathPrefix {
-        /// The Janus aggregator API will be served on the same address as the DAP API, but relative
-        /// to the provided prefix. e.g., if `path_prefix` is `aggregator-api`, then the DAP API's
-        /// uploads endpoint would be `{listen-address}/tasks/{task-id}/reports`, while task IDs
-        /// could be obtained from the aggregator API at `{listen-address}/aggregator-api/task_ids`.
-        path_prefix: String,
-        /// Resource location at which the DAP service managed by this aggregator api can be found
-        /// on the public internet. Required.
-        public_dap_url: Url,
-    },
+#[serde(deny_unknown_fields)]
+pub struct AggregatorApi {
+    /// Address on which this server should listen for connections to the Janus aggregator API
+    /// and serve its API endpoints, independently from the address on which the DAP API is
+    /// served.
+    listen_address: Option<SocketAddr>,
+    /// The Janus aggregator API will be served on the same address as the DAP API, but relative
+    /// to the provided prefix. e.g., if `path_prefix` is `aggregator-api`, then the DAP API's
+    /// uploads endpoint would be `{listen-address}/tasks/{task-id}/reports`, while task IDs
+    /// could be obtained from the aggregator API at `{listen-address}/aggregator-api/task_ids`.
+    path_prefix: Option<String>,
+    /// Resource location at which the DAP service managed by this aggregator api can be found
+    /// on the public internet. Required.
+    public_dap_url: Url,
 }
 
-impl AggregatorApi {
-    fn public_dap_url(&self) -> &Url {
-        match self {
-            AggregatorApi::ListenAddress { public_dap_url, .. } => public_dap_url,
-            AggregatorApi::PathPrefix { public_dap_url, .. } => public_dap_url,
+fn deserialize_aggregator_api<'de, D>(deserializer: D) -> Result<Option<AggregatorApi>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let aggregator_api: Option<AggregatorApi> =
+        serde_yaml::from_value(Deserialize::deserialize(deserializer)?)
+            .map_err(de::Error::custom)?;
```

Comment:
This can be simplified by using the deserializer directly. (ideally, we'd like the body of Serialize and Deserialize implementations to be independent of any serializer or deserializer)
```suggestion
    let aggregator_api: Option<AggregatorApi> = Deserialize::deserialize(deserializer)?;
```